lib/cocoapods/installer.rb

```diff
@@ -206,7 +221,15 @@ def create_generator(generate_multiple_pod_projects = false)
 
     # Generates the Xcode project(s) that go inside the `Pods/` directory.
     #
-    def generate_pods_project(generator = create_generator(installation_options.generate_multiple_pod_projects))
+    def generate_pods_project
+      stage_sandbox(sandbox, pod_targets)
+      clean_sandbox(pod_targets, aggregate_targets)
+      create_and_save_projects
+      write_lockfiles
+      SandboxDirCleaner.new(sandbox, pod_targets, aggregate_targets).clean!
```

Comment:
nit: it's a bit odd that _some_ of the cleaning happens in `clean_sandbox` and some of it happens through `SandboxDirCleaner` - is there any way to consolidate that? I'm not familiar with how the two code paths differ

Reply:
Yeah, so `clean_sandbox` cleans up the directories for pod targets that will be generated and `SandboxDirCleaner` handles cleaning up directories for pod targets that were removed.

Reply:
`clean_sandbox` and `SandboxDirCleaner` will diverge with incremental installation since `clean_sandbox` will take in the list of generated pod targets while `SandboxDirCleaner` will always take in all pod and aggregate targets in order to compute what needs to be removed.

Reply:
ah okay that makes sense. a `SandboxDirPreparer` or something would be cool but not necessary for this PR 👍